lib/fluent/plugin/out_vmware_loginsight.rb

```diff
@@ -119,13 +128,10 @@ def shorten_key(key)
         # LI doesn't allow some characters in field 'name'
         # like '/', '-', '\', '.', etc. so replace them with @flatten_hashes_separator
         key = key.gsub(/[\/\.\-\\]/,@flatten_hashes_separator).downcase
-        # shorten field names
-        key = key.gsub(/kubernetes_/,'k8s_')
-        key = key.gsub(/namespace/,'ns')
-        key = key.gsub(/labels_/,'')
-        key = key.gsub(/_name/,'')
-        key = key.gsub(/_hash/,'')
-        key = key.gsub(/container_/,'')
+        # shorten field names using provided shorten_keys parameters
+        shorten_keys.each do | match, replace |
```

Comment:
`@shorten_keys`

Reply:
Did test in my environment, works as expected

Tried some error condition, invalid hash map cause fluent to not start whereas invalid value (array instead of string) causes log to not be sent

input: shorten_keys {  :"", "_hash":""  }
error: `parse_error!': got incomplete JSON hash configuration at fluent.conf line 135,9 (Fluent::ConfigParseError)
symphony-logging-daemonset-57prb                   0/1     CrashLoopBackOff

input: shorten_keys { "kubernetes_":["k8s_","kubernetes_",""] }
error: #0 failed to emit fluentd's log event tag="fluent.error" event={"error"=>"#<TypeError: no implicit conversion of Array into String>", "message"=>"failed to flush timeout buffer error_class=TypeError error=\"no implicit conversion of Array into String\""} error_class=TypeError error="no implicit conversion of Array into String"
symphony-logging-daemonset-rjjm4                   1/1     Running